File: src/export_response.js
const { Pool } = require("pg");
const fs = require("fs");
const fastcsv = require("fast-csv");
require("dotenv").config();
const pool = new Pool({
  user: process.env.DB_USER,
  host: process.env.DB_HOST,
  database: process.env.DB_NAME,
  password: process.env.DB_PASSWORD,
  port: process.env.DB_PORT || 5432,
});

const exportToCSV = async () => {
  try {
    const result = await pool.query('SELECT * FROM responses'); // Change table name as needed

    const ws = fs.createWriteStream('/data/horse/ws/kyim310f-cms-pro/surve-app/result/sample.csv');
    
    fastcsv
      .write(result.rows, { headers: true })
      .pipe(ws);

    console.log("Data has been written to the CSV file successfully.");
  } catch (error) {
    console.error("Error exporting data to CSV:", error);
  }
};

exportToCSV();

File: src/export_response.py
import psycopg2
import pandas as pd

# Database connection details
DB_HOST = "localhost"  # Replace with your host
DB_PORT = "5432"       # Replace with your port (default is 5432)
DB_NAME = "survey_db"  # Your database name
DB_USER = "postgres"  # Replace with your DB username
DB_PASSWORD = "postgres"  # Replace with your DB password

# Initialize the connection variable
conn = None

# Connect to the PostgreSQL database
try:
    conn = psycopg2.connect(
        host=DB_HOST,
        port=DB_PORT,
        dbname=DB_NAME,
        user=DB_USER,
        password=DB_PASSWORD
    )
    print("Connection to the database established successfully.")
except Exception as e:
    print("Error: Unable to connect to the database")
    print(e)

# Only proceed if the connection was successful
if conn:
    # Query to fetch all data from the responses table
    query = "SELECT * FROM responses;"

    # Exporting the data to CSV
    try:
        # Use pandas to execute the query and load data into a DataFrame
        df = pd.read_sql_query(query, conn)

        # Export the DataFrame to CSV
        df.to_csv("responses_export.csv", index=False)
        print("Data has been exported to responses_export.csv successfully.")
    except Exception as e:
        print("Error exporting data to CSV:")
        print(e)
    finally:
        # Close the database connection
        conn.close()
        print("Database connection closed.")
else:
    print("Database connection was not established, so no data was exported.")
